src/command_modules/azure-cli-servicebus/azure/cli/command_modules/servicebus/_params.py

```diff
@@ -0,0 +1,173 @@
+            c.argument('max_delivery_count', type=int, help='Number of maximum deliveries.')
+            c.argument('status', arg_type=get_enum_type(['Active', 'Disabled', 'SendDisabled', 'ReceiveDisabled']))
+            c.argument('enable_batched_operations', arg_type=get_three_state_flag(), help='A boolean value that indicates whether server-side batched operations are enabled.')
+            c.argument('auto_delete_on_idle', validator=_validate_auto_delete_on_idle, options_list=['--auto-delete-on-idle'], help='ISO 8601 timeSpan  or duration time format for idle interval after which the topic is automatically deleted. The minimum duration is 5 minutes.')
+            c.argument('forward_to', help='Queue/Topic name to forward the messages')
+            c.argument('forward_dead_lettered_messages_to', help='Queue/Topic name to forward the Dead Letter message')
+
+    # Region Subscription Rules
+    # Rules Create
+
+    with self.argument_context('servicebus topic subscription rule') as c:
+        c.argument('rule_name', arg_type=name_type, completer=get_rules_command_completion_list, help='Name of Rule')
+        c.argument('subscription_name', options_list=['--subscription-name'], id_part='child_name_2', completer=get_subscriptions_command_completion_list, help='Name of Subscription')
+        c.argument('topic_name', options_list=['--topic-name'], id_part='child_name_1', completer=get_topic_command_completion_list, help='Name of Topic')
+
+    for scope in ['servicebus topic subscription rule create', 'servicebus topic subscription rule update']:
+        with self.argument_context(scope) as c:
+            c.argument('action_sql_expression', arg_group='action', help='Action SQL expression.')
+            c.argument('action_compatibility_level', arg_group='action', type=int, help='This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20.')
+            c.argument('action_requires_preprocessing', options_list=['--enable-action-preprocessing'], arg_group='action', arg_type=get_three_state_flag(), help='A boolean value that indicates whether the rule action requires preprocessing.')
+            c.argument('filter_type', arg_type=get_enum_type(FilterType), help='Rule Filter types')
+            c.argument('filter_sql_expression', arg_group='sql_filter', help='SQL expression. e.g.')
+            c.argument('filter_requires_preprocessing', options_list=['--enable-sql-preprocessing'], arg_group='sql_filter', arg_type=get_three_state_flag(), help='A boolean value that indicates whether the rule action requires preprocessing.')
+            c.argument('action_compatibility_level', type=int, help='This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20.')
+            c.argument('correlation_id', arg_group='correlation_filter', help='Identifier of correlation.')
```

Comment:
arg_groups should appear like this "Correlation Filter" not like this "correlation_filter" as this is how they are displayed in help.  Also, you should group these in an argument context:

```Python
with self.argument_context(scope, arg_group='Correlation Filter') as c:
   c.argument('message_id', 'help='...')
   c.argument('to', help='...')
```

This way you don't have to repeat "arg_group='...'" over and over and it makes the blocks more readable in the code.

Reply:
implemented